### ISSUE TITLE
Support password protected stories

### ISSUE THREAD

**Issue by swissspidy** [GoogleForCreators/web-stories-wp#2335]
## Feature Description

WordPress provides the ability to protect content with a (simple) password.

As per https://github.com/google/web-stories-wp/issues/87#issuecomment-577026598https://github.com/google/web-stories-wp/issues/87#issuecomment-577026598 and #405 we might want to enable this for stories too. Quote:

> Password and privacy features make sense, but I wouldn't give them super high priority. If they're easy to implement, I'd say let's do it. Otherwise defer to later.

Implementing this would require showing the password page in the current theme, and un-hiding the password field in the document panel.

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Implementation Brief

* Enable password field in `packages/story-editor/src/components/panels/document/status/status.js`
* In `\Google\Web_Stories\Renderer\Single::filter_template_include()`, don't filter the template if `post_password_required()` returns `true`

**Comment by telizarose** [comment]
I'll be the first to validate this as a feature is probably needed. It was actually the first thing I looked for when I installed beta and I came here to see if it was planned. Happy to see it up here!

**Some Use Cases:** 
- Writers (like me) who want to password protect some stories to make them only viewable after a purchase or donation has been made (the password would be sent in the receipt for the purchase) 
- High schools and other kinds of youth-based websites (sports clubs, etc) that might publish stories with photos of minors and can provide more privacy and protection by giving parents and kids the password to stories published. 

You might not need to design a password page if the native WordPress password protection functionality that's used in posts works with stories (I'm not sure - not a developer).
